python/hail/methods/statgen.py

```diff
@@ -3027,12 +3033,12 @@ def ld_prune(call_expr, r2=0.2, bp_window_size=1000000, memory_per_core=256, kee
         0.0)
 
     std_gt_bm = BlockMatrix.from_entry_expr(standardized_mean_imputed_gt_expr, block_size)
+    r2_bm = (std_gt_bm @ std_gt_bm.T) ** 2
 
-    locally_pruned_positions = locally_pruned_table.key_by(contig=locally_pruned_table.locus.contig)
-    locally_pruned_positions = locally_pruned_positions.select(pos=locally_pruned_positions.locus.position)
+    _, stops = hl.locus_windows(locally_pruned_table, bp_window_size)
```

Comment:
Why don't you need the starts?

Reply:
Because we only want to compute / represent each pair of variants once. Using the (super)diagonal for starts forces the earlier variant to be first in the pair.